R/deleteGates.R

```diff
@@ -0,0 +1,39 @@
+#' Delete gate
+#'
+#' Deletes a gate or a tailored gate family
+#' Compound gates (quadrant and split) are made up of "sectors." 
+#' Quadrant gates have four sectors (upper-right, upper-left, lower-left, lower-right) and split gates have two sectors (left and right).
+#' In addition to the top-level GID (like simple gates), these gates have model.gids and names lists that specify the GID and name for each sector, in the order shown above. 
+#' Populations using compound gates must reference these sector GIDs; referencing the top-level GID of a compound gate is meaningless.
+#' 
+#'
+#' @param experimentId ID of experiment.
+#' @param gid ID of gate family.
+#' @param gateId ID of gate.
+#' @param exclude Gate ID to exclude from deletion.
+#' @export
+#' @examples
+#' \dontrun{
+#' deleteGate(experimentId, gateId)
+#' }
+#' 
+#' TODO:(ge) This needs to account for compound gates (Quadrant, Split)
+
+deleteGates = function(experimentId, gid, gateId = NULL, exclude = NULL) {
```

Comment:
This one's a little tricky. I'd probably make `gid = NULL` also, and then structure the `if`s like

```R
if (!is.null(gateId)) {
  # what you have
} else if (!is.null(gid)) {
  base = # what you have, plus ?gid=%s
  if (!is.null(exclude)) { # the 'exclude' parameter requires using the 'gid' parameter, so it's nested here
    # add &exclude=
  }
  baseDelete(url)
}
```

I'd also check that only one of `gid` or `gateId` are not null; supplying both is a user error.